tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/SignificantIndentationSuite.scala

```diff
@@ -0,0 +1,363 @@
+              Term.Name("fx"),
+              Nil,
+              List(List()),
+              Some(Type.Name("Unit")),
+              Term.Block(Nil)
+            ),
+            Defn.Def(
+              List(Mod.Private(Name(""))),
+              Term.Name("f2"),
+              Nil,
+              Nil,
+              Some(Type.Name("Int")),
+              Lit.Int(1)
+            )
+          )
+        )
+      )
+    )
+  }
+
+  test("selftype-class") {
+    val code = """|class A extends B:
+                  |  thisPhase =>
+                  |  expr1
+                  |  expr2
+                  |""".stripMargin
+    val output = """|class A extends B { thisPhase =>
+                    |  expr1
+                    |  expr2
+                    |}
+                    |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = Some(output))(
+      Defn.Class(
+        Nil,
+        Type.Name("A"),
+        Nil,
+        Ctor.Primary(Nil, Name(""), Nil),
+        Template(
+          Nil,
+          List(Init(Type.Name("B"), Name(""), Nil)),
+          Self(Term.Name("thisPhase"), None),
+          List(Term.Name("expr1"), Term.Name("expr2"))
+        )
+      )
+    )
+  }
+
+  test("lambda-method") {
+    val code = """|object X:
+                  |  val fn = (pa, pb) =>
+                  |    def helper = 3
+                  |    3
+                  |  end fn
+                  |""".stripMargin
+    val output = """|object X {
+                    |  val fn = (pa, pb) => {
+                    |    def helper = 3
+                    |    3
+                    |  }
+                    |  end fn
+                    |}
+                    |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = Some(output))(
+      Defn.Object(
+        Nil,
+        Term.Name("X"),
+        Template(
+          Nil,
+          Nil,
+          Self(Name(""), None),
+          List(
+            Defn.Val(
+              Nil,
+              List(Pat.Var(Term.Name("fn"))),
+              None,
+              Term.Function(
+                List(
+                  Term.Param(Nil, Term.Name("pa"), None, None),
+                  Term.Param(Nil, Term.Name("pb"), None, None)
+                ),
+                Term.Block(
+                  List(Defn.Def(Nil, Term.Name("helper"), Nil, Nil, None, Lit.Int(3)), Lit.Int(3))
+                )
+              )
+            ),
+            Term.EndMarker(Term.Name("fn"))
+          )
+        )
+      )
+    )
+  }
+
+  test("case-for-in-match") {
+    val code = """|class A {
+                  |    def forward: Unit = parents match
+                  |      case a =>
+                  |        for case a: TP <- body do
+                  |          fordo
+                  |      case b => ok
+                  |
+                  |    private def transformAnnot: Tree
+                  |}
+                  |""".stripMargin
+    runTestAssert[Stat](code, assertLayout = None)(
```

Comment:
Same here, could you add assert layout if possible?

Reply:
ok